File: dart/lib/src/base/models/square.dart
import "dart:math";

import "package:shapes/base/commons.dart";

class Square implements Dimensional {
  double? _side;

  Square(double side) {
    double dimOne = side > 0 ? side : 1;
    _side = dimOne;
  }

  @override
  double area() {
    return pow(_side!, 2).toDouble();
  }

  @override
  double perimeter() {
    return 4 * _side!;
  }
}

File: typescript/lib/base/models/square.ts
import { Dimensional } from "../commons";

export class Square implements Dimensional {
    private side: number;

    constructor(side: number) {
        const dimOne: number = side > 0 ? side : 1;
        this.side = dimOne;
    }

    area(): number {
        return Math.pow(this.side, 2);
    }

    perimeter(): number {
        return 4 * this.side;
    }
}
